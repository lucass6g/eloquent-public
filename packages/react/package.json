{
  "name": "@eloquent/react",
  "version": "0.0.1",
  "description": "",
  "main": "./dist/index.js",
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.js",
      "require": "./dist/index.js",
      "types": "./dist/index.d.ts"
    }
  },
  "scripts": {
    "build": "tsup"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "tsup": "^8.0.2",
    "@types/react": "^18.2.66",
    "@types/react-dom": "^18.2.22",
    "react": "^18.2.0",
    "typescript": "^5.4.2"
  }
}
