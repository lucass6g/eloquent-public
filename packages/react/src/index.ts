export {Switch, Case, Default} from './components/conditional-switch'