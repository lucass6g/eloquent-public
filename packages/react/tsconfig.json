{
  "extends": "../../packages/config/base.tsconfig.json",
  "include": ["src"],
  "exclude": ["dist", "build", "node_modules"],
  "compilerOptions": {
    "forceConsistentCasingInFileNames": false,
    "rootDir": "src",
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
