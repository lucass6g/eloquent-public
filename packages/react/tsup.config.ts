import { defineConfig } from 'tsup';

export default defineConfig({
	name: "@eloquent/react",
	entry: ["src/index.ts"],
	outDir: "dist",
	target: "es2022",
	format: ["cjs", "esm"],
	splitting: false,
	sourcemap: true,
	clean: true,
	dts: true,
	external: ["react", 'react-dom', 'react/jsx-runtime'],
	cjsInterop: true,
	treeshake: true,
})