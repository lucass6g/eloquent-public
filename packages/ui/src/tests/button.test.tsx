import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Button } from "..";
import { describe, expect, it, vi } from "vitest";

describe("Button", () => {
  describe("should match the snapshot", () => {
    it("default", () => {
      const component = render(<Button>button</Button>);

      expect(component.baseElement).toMatchSnapshot();
    });

    it("secondary", () => {
      const component = render(<Button variant="secondary">button</Button>);

      expect(component.baseElement).toMatchSnapshot();
    });

    it("outline", () => {
      const component = render(<Button variant="secondary">button</Button>);

      expect(component.baseElement).toMatchSnapshot();
    });

    it("destructive", () => {
      const component = render(<Button variant="destructive">button</Button>);

      expect(component.baseElement).toMatchSnapshot();
    });

    it("disabled", () => {
      const component = render(<Button disabled>button</Button>);

      expect(component.baseElement).toMatchSnapshot();
    });

    it("ghost", () => {
      const component = render(<Button variant="secondary">button</Button>);

      expect(component.baseElement).toMatchSnapshot();
    });

    it("link", () => {
      const component = render(<Button variant="link">button</Button>);

      expect(component.baseElement).toMatchSnapshot();
    });

    it("icon", () => {
      const component = render(<Button size="icon">+</Button>);

      expect(component.baseElement).toMatchSnapshot();
    });
  });

  it("should click correctly", async () => {
    const clickFn = vi.fn();
    render(<Button onClick={clickFn}>button</Button>);

    const button = screen.getByRole("button", { name: /button/i });
    await waitFor(() => userEvent.click(button));

    expect(clickFn).toHaveBeenCalled();
  });

  it("should be disabled", async () => {
    const clickFn = vi.fn();
    render(
      <Button disabled onClick={clickFn}>
        button
      </Button>
    );

    const button = screen.getByRole("button", { name: /button/i });
    await waitFor(() => userEvent.click(button));

    expect(button).toBeDisabled();
    expect(clickFn).not.toHaveBeenCalled();
  });
});
