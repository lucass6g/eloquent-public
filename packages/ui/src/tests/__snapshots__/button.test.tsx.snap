// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > should match the snapshot > default 1`] = `
<body>
  <div>
    <button
      class="inline-flex items-center justify-center whitespace-nowrap rounded-lg text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground shadow hover:bg-green-dark-400 h-9 px-4 py-2 border-lg"
    >
      button
    </button>
  </div>
</body>
`;

exports[`Button > should match the snapshot > destructive 1`] = `
<body>
  <div>
    <button
      class="inline-flex items-center justify-center whitespace-nowrap rounded-lg text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 disabled:pointer-events-none disabled:opacity-50 bg-destructive text-destructive-foreground shadow-sm hover:opacity-90 h-9 px-4 py-2 border-lg"
    >
      button
    </button>
  </div>
</body>
`;

exports[`Button > should match the snapshot > disabled 1`] = `
<body>
  <div>
    <button
      class="inline-flex items-center justify-center whitespace-nowrap rounded-lg text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground shadow hover:bg-green-dark-400 h-9 px-4 py-2 border-lg"
      disabled=""
    >
      button
    </button>
  </div>
</body>
`;

exports[`Button > should match the snapshot > ghost 1`] = `
<body>
  <div>
    <button
      class="inline-flex items-center justify-center whitespace-nowrap rounded-lg text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent text-accent-foreground h-9 px-4 py-2 border-lg"
    >
      button
    </button>
  </div>
</body>
`;

exports[`Button > should match the snapshot > icon 1`] = `
<body>
  <div>
    <button
      class="inline-flex border-lg items-center justify-center whitespace-nowrap rounded-lg text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground shadow hover:bg-green-dark-400 h-9 w-9 first-child:stroke-primary-foreground"
    >
      +
    </button>
  </div>
</body>
`;

exports[`Button > should match the snapshot > link 1`] = `
<body>
  <div>
    <button
      class="inline-flex items-center justify-center whitespace-nowrap rounded-lg text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 disabled:pointer-events-none disabled:opacity-50 text-primary underline-offset-4 hover:underline h-9 px-4 py-2 border-lg"
    >
      button
    </button>
  </div>
</body>
`;

exports[`Button > should match the snapshot > outline 1`] = `
<body>
  <div>
    <button
      class="inline-flex items-center justify-center whitespace-nowrap rounded-lg text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 disabled:pointer-events-none disabled:opacity-50 shadow-sm border text-accent-foreground hover:bg-accent hover:text-accent-foreground h-9 px-4 py-2 border-lg"
    >
      button
    </button>
  </div>
</body>
`;

exports[`Button > should match the snapshot > secondary 1`] = `
<body>
  <div>
    <button
      class="inline-flex items-center justify-center whitespace-nowrap rounded-lg text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 disabled:pointer-events-none disabled:opacity-50 bg-secondary text-secondary-foreground shadow-sm hover:opacity-90 h-9 px-4 py-2 border-lg"
    >
      button
    </button>
  </div>
</body>
`;
